tests/FakeItEasy.Specs/DummyCreationSpecs.cs

```diff
@@ -805,6 +834,23 @@ public LazilyCreated()
             }
         }
         #pragma warning restore CA1052
+
+        public class ClassWhoseDummyIsNull
+        {
+        }
+
+        public class ClassWhoseDummyIsNullFactory : DummyFactory<ClassWhoseDummyIsNull>
+        {
+            protected override ClassWhoseDummyIsNull Create() => null;
+        }
+
+        public class ClassThatRequiresClassWhoseDummyIsNull
+        {
+            [SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", MessageId = "unused", Justification = "This is just a dummy argument.")]

```

Comment:
I think we can use `#pragma warning disable CA1801` now. It's fine to leave it as is for now, but we could consider switching to the new syntax in the future. The only benefit of the attribute approach is that it lets us give a justification for the suppression, but we could just do this in a comment.

Reply:
You prefer the pragma because it's terser? Any other reason? I assume we'd want to restore after, or are you thinking of a blanket one for the file?

Reply:
Yes, because it's terser. But now that I think about it, if we restore it immediately afterwards, it makes things even worse... If only we had https://github.com/dotnet/csharplang/issues/1070!

Reply:
Yeah, the's the one thing I missed when switching from C++ to C#. (What you describe is very nearly "suppress" from https://docs.microsoft.com/en-us/cpp/preprocessor/warning?view=vs-2019.)